patroni/ha.py

```diff
@@ -122,6 +123,15 @@ def bootstrap(self):
 
     def recover(self):
         self.recovering = True
+        if self.has_lock() and self.update_lock():
```

Comment:
`self.recovering`  should be set to True just before calling `self.follow`.
Otherwise we have following situation:
```
2016-09-13 12:20:53,994 INFO: Master crashed. Failing over.
2016-09-13 12:20:54,004 INFO: Leader key released
2016-09-13 12:20:54,004 INFO: stopped PostgreSQL to fail over after a crash
2016-09-13 12:20:54,239 INFO: Lock owner: None; I am dbnode3
2016-09-13 12:20:54,239 INFO: not healthy enough for leader race
2016-09-13 12:21:04,014 INFO: New leader detected, proceeding with recovery.
localhost:5432 - no response
2016-09-13 12:21:04,257 INFO: Lock owner: dbnode2; I am dbnode3
2016-09-13 12:21:04,258 INFO: failed to start postgres  # But we didn't even tried!
localhost:5432 - no response
2016-09-13 12:21:14,049 WARNING: Postgresql is not running.
```